.github/workflows/registry_upload.yml

```diff
@@ -0,0 +1,90 @@
+name: Create and publish a Docker image
+
+# Configures this workflow to run on main when testing CI is completed.
+on:
+  workflow_run:
+    workflows: ["CI"]
+    branches: [main]
+    types:
+      - completed
+
+# Defines two custom environment variables for the workflow. These are used for the Container registry domain, and a name for the Docker image that this workflow builds.
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+  CPU: "arm64"
+
+jobs:
+
+  on-failure:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.workflow_run.conclusion == 'failure' }}
+    steps:
+      - run: echo 'The triggering CI workflow failed'
+  docker-build-and-publish:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    # Sets the permissions granted to the `GITHUB_TOKEN` for the actions in this job.
+    permissions:
+      contents: read
+      packages: write
+    strategy:
+      matrix:
+        include:
+          - context: ./datastore/datastore/
+            IMAGE_NAME: datastore
+            CPU: ["arm64", "amd64"]
```

Comment:
I think these `CPU` can be dropped here, as they are only used in line 50, where the default `CPU` definition from line 15 is fine (amd64 is automatically available).

Reply:
Yeah, this line is never used either